commands/login.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) OpenFaaS Project 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package commands
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/openfaas/faas-cli/config"
+	"github.com/spf13/cobra"
+)
+
+var (
+	username      string
+	password      string
+	passwordStdin bool
+)
+
+func init() {
+	loginCmd.Flags().StringVar(&gateway, "gateway", defaultGateway, "Gateway URI")
+	loginCmd.Flags().StringVarP(&username, "username", "u", "", "Gateway username")
+	loginCmd.Flags().StringVarP(&password, "password", "p", "", "Gateway password")
+	loginCmd.Flags().BoolVar(&passwordStdin, "password-stdin", false, "Reads the gateway password from stdin")
```

Comment:
We can tweak the UX after merging but password-stdin should be as easy to use as -p so that people don't just default to having their password in the bash history.